cocotb/log.py

```diff
@@ -86,18 +89,23 @@ def default_config():
 
     # apply level settings for cocotb
     log = logging.getLogger('cocotb')
-    level = os.getenv("COCOTB_LOG_LEVEL", "INFO")
+
+    # All log levels are upper case, convert the user input for convenience.
+    level = os.getenv("COCOTB_LOG_LEVEL", logging.getLevelName(_DEFAULT_LOG_LEVEL)).upper()
+
     try:
-        _default_log = getattr(logging, level)
-    except AttributeError:
-        log.error("Unable to set logging level to %r" % level)
-        _default_log = logging.INFO
-    log.setLevel(_default_log)
+        log.setLevel(level)
+    except ValueError as e:
+        valid_levels = ('CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG')
+        log.error("Invalid log level %r passed through the COCOTB_LOG_LEVEL "
+                  "environment variable, using %r instead. Valid log levels: %s"
+                  % (level, logging.getLevelName(_DEFAULT_LOG_LEVEL),
+                     ', '.join(valid_levels)))
```

Comment:
```suggestion
                  "environment variable, using %r instead. Valid log levels: %s",
                  level, logging.getLevelName(_DEFAULT_LOG_LEVEL),
                  ', '.join(valid_levels)))
```
No need to use `%` here, the logging module does it for you